src/common/file_retriever.py

```diff
@@ -45,17 +47,31 @@ def _s3_save(protocol, source_loc, dest_obj, auth="saml"):
     :type source_loc: str
     :param dest_obj: Receives the data downloaded from the source file
     :type dest_obj: File-like object
-    :param auth: optional S3 auth profile (defaults to "saml")
+    :param auth: optional S3 auth profile (defaults to all available profiles)
     :type auth: str
 
     :returns: None, the data goes to dest_obj
     """
+    if auth is not None:
+        aws_profiles = auth
+    else:
+        aws_profiles = boto3.Session().available_profiles + [None]
+
     bucket, key = source_loc.split("/", 1)
-    s3 = boto3.session.Session(profile_name=auth).resource("s3")
-    s3.Object(bucket, key).download_fileobj(dest_obj)
 
+    for profile in aws_profiles:
+        try:
+            logger.info("Trying auth profile '%s'", profile)
+            s3 = boto3.session.Session(profile_name=profile).resource("s3")
+            s3.Object(bucket, key).download_fileobj(dest_obj)
+            return
+        except botocore.exceptions.NoCredentialsError:
+            pass
+
+    raise botocore.exceptions.NoCredentialsError()  # never got the file
 
-def _web_save(protocol, source_loc, dest_obj, auth=None):
+
+def _web_save(protocol, source_loc, dest_obj, auth, logger):
```

Comment:
Since requests.get takes auth as a kwarg, couldn't it be a kwarg in _web_save too?